cogs5e/utils/checkutils.py

```diff
@@ -136,8 +138,15 @@ def _run_common(skill, args, embed, mod_override=None, rr_format="Check {}"):
         desc_out.append(f"**DC {dc}**")
 
     for i in range(iterations):
-        # advantage
-        adv = args.adv(boolwise=True, ephem=True)
+        # ieffect handling
+        # Combine args/ieffect advantages - adv/dis (#1552)
+        if isinstance(caster, init.Combatant):
+            adv = reconcile_adv(
+                adv = args.last('adv', type_=bool, ephem=True) or stat in caster.active_effects('sadv'),
+                dis = args.last('dis', type_=bool, ephem=True) or stat in caster.active_effects('sdis'))
+        else:
+            adv = args.adv(boolwise=True, ephem=True)
```

Comment:
From Discord:
> :\ So _run_common takes in the full args but not the context. Can't think of a way to do the reconcile there, which means it needs to be in run_save instead.
> Since the args get dropped on the floor after _run_common, is it okay to just insert them like they are now?
> Wait- no it doesn't. Dangit.
>The easiest answer I can think of is passing the caster down one more level.

Well, we're already modifying args in L108... you're right that the args don't get reused in outer scopes on checks, so I think modifying args is fine. It seems a little more elegant than having the `stat` argument be kind of ambiguous and having to pass the caster down a level, anyway

Reply:
Got it will return to the version that just appends as it's much simpler.